### PR TITLE
Cmake plugins and install things

### PR DESCRIPTION
**Pull request by cmcfarlen**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -183,3 +190,6 @@ add_subdirectory(mgmt/utils)
 add_subdirectory(mgmt/config)
 add_subdirectory(mgmt/rpc)
 add_subdirectory(src/traffic_server)
+add_subdirectory(plugins)
+
+add_subdirectory(configs)
```
Comment (bneradt) [inline CMakeLists.txt:195]: https://stackoverflow.com/a/44704969/629530

**Review by bneradt — CHANGES_REQUESTED** [review]

File: plugins/cache_promote/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+#  the License.
+#
+#######################
+
+add_atsplugin(cache_promote
+        cache_promote.cc
+        configs.cc
+        policy.cc
+        lru_policy.cc
+        policy_manager.cc
```
Comment (bneradt) [inline plugins/cache_promote/CMakeLists.txt:23]: Out of curiosity: how do header files relate to this? Will CMake detect changes in header files and rebuild the required files?

**Comment by bneradt** [comment]
This is great progress. Thanks @cmcfarlen for working on this.

File: CMakeLists.txt (hunk 1 above)
Comment (cmcfarlen) [reply CMakeLists.txt:195]: I had this for vim, but been using clion lately.  I'll look up the equivalent.

**Comment by ywkaras** [comment]
How does CMake handle installing headers for externally-built plugings?  As done here in Makefile.am:  https://github.com/apache/trafficserver/blob/master/include/tscpp/api/Makefile.am .

File: plugins/cache_promote/CMakeLists.txt (hunk 2 above)
Comment (cmcfarlen) [reply plugins/cache_promote/CMakeLists.txt:23]: Yes! I did an experiment where I just changed a random header file.  It recompiled .cc files that included that, recreated libraries and then rebuild other targets that depended on that target (even transitive ones)

**Review by bneradt — APPROVED** [review]

**Pull request merged**